pandas/core/base.py

```diff
@@ -804,9 +804,9 @@ def argmax(
         dtype: float64
 
         >>> s.argmax()
-        2
+        np.int64(2)
         >>> s.argmin()
-        0
```

Comment:
There are a few of these where I'm wondering if we should be returning Python scalars instead of NumPy. Should issues be opened for these?

cc @pandas-dev/pandas-core

Reply:
I think generally we always want to return Python scalars (IIRC we got a lot of issues about this in iteration and iteration-like APIs in the past)

Reply:
Even just wrapping the result of `Series._reduce` with `maybe_box_naive` breaks 692 tests. From a cursory look, they're tests that are expecting a NumPy scalar back. A lot however are something like `op(data).any().any()` so that they will work with DataFrame and Series. I plan to bring this up in the next dev meeting.

Reply:
I agree we should always return Python scalars. I'm surprised at the amount of failures that expect NumPy scalars

Reply:
We could put it up behind a `future` option, maybe something like `always_return_python_scalars`